### DISCUSSION TITLE
Proper configuration for communication across different machines

### DISCUSSION THREAD

**Discussion by dvucko** [OpenDDS/OpenDDS#3635] [Q&A] [answered]
I have a publisher (writer) and a subscriber (reader), each an independent process. The subscriber normally receives message from the publisher when run on the same machine, but the attempts to get publishing to work when the two are running on different machines failed. Both machines normally see each other and are in the same domain.

When run locally, both subscriber and publisher use the following `OpenDDS.ini`:
```
[common]
DCPSDebugLevel=0
DCPSDefaultDiscovery=the_rtps_discovery
DCPSGlobalTransportConfig=the_config
DCPSDefaultAddress=127.0.0.1
DCPSPendingTimeout=4

[config/the_config]
transports=the_transport

[transport/the_transport]
transport_type=rtps_udp
heartbeat_period=1 

[rtps_discovery/the_rtps_discovery]
ResendPeriod=5
SedpLocalAddress=127.0.0.1:
SpdpLocalAddress=127.0.0.1
```

Assuming IPs `publisherMachineIP` and `subscriberMachineIP`, what would the proper config file look like? I've obviously checked the documentation, but being new to the terminology I'm not quite sure.

**Comment by simpsont-oci** [comment, marked as answer]
You probably want to use the outward-facing IP (not 127.0.0.1) for the local address, if you think you need to specify it. In that case, you'd need two different config files.

Generally, you can simply not specify DCPSDefaultAddress or SedpLocalAddress / SpdpLocalAddress and OpenDDS will attempt to bind all available interfaces, in which case the config files would probably be the same.

Another thing to watch out for when moving to two different machines is your firewall, which may need to be modified or disabled to allow RTPS traffic, as well as making sure you're on a network that allows for multicast traffic (or manually specifying which remote addresses to send discovery information to using SpdpSendAddrs). If you suspect your firewall is interfering, you can use something like wireshark or tcpdump to capture udp traffic and look for RTPS packets.

**Reply by dvucko** [reply]
Thank you for your suggestions. While firewall was disabled all along, this indeed seems to be a network configuration issue based on some investigation. I'll try to post something more concrete later.

**Reply by dvucko** [reply]
Simply not specifying any IPs and making sure multicast works was enough to get this working, as pointed out.
Minimal config that works:
```
[common]
DCPSDebugLevel=0
DCPSDefaultDiscovery=DEFAULT_RTPS

[transport/the_rtps_transport]
transport_type=rtps_udp
```

**Reply by simpsont-oci** [reply]
Glad to hear you got it working!
